tests/integration/pilot/custom_gateway_test.go

```diff
@@ -0,0 +1,222 @@
+// +build integ
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/protocol"
+	"istio.io/istio/pkg/test/env"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/istioctl"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/image"
+	"istio.io/istio/pkg/test/shell"
+	"istio.io/istio/pkg/test/util/retry"
+	"istio.io/istio/tests/integration/pilot/common"
+	"istio.io/istio/tests/util"
+)
+
+const (
+	customServiceGateway = "custom-ingressgateway"
+	iopGWTemplate        = `
+apiVersion: install.istio.io/v1alpha1
+kind: IstioOperator
+metadata:
+  name: custom-ingressgateway-iop
+  namespace: %s
+spec:
+  profile: empty
+  components:
+    ingressGateways:
+    - name: custom-ingressgateway
+      label:
+        istio: custom-ingressgateway
+      namespace: %s
+      enabled: true
+`
+	gwTemplate = `
+apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: %s
+spec:
+  selector:
+    istio: %s
+  servers:
+  - port:
+      number: 80
+      name: http
+      protocol: HTTP
+    hosts:
+    - "*"
+`
+	vsTemplate = `
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: %s
+spec:
+  hosts:
+  - "*"
+  gateways:
+  - %s
+  http:
+  - match:
+    - uri:
+        prefix: /
+    route:
+    - destination:
+        host: %s
+`
+)
+
+var (
+	customGWNamespace namespace.Instance
+	cgwInst           istio.Instance
+	ManifestPath      = filepath.Join(env.IstioSrc, "manifests")
+)
+
+// TestAccessAppViaCustomGateway tests access to an application using a custom gateway
```

Comment:
The TestAccessAppViaCustomGateway seems to run in 3s in CI. I don't see how that is possible - it seems like its perhaps not actually running at all?

Reply:
I'll look into that one. I noted the 3 seconds, thought wow, it is faster if I put it in another suite that spun up the control plane and the apps (as you commented earlier). Now the test just creates the gateway and verifies connectivity.